reader_concurrency_semaphore.cc

```diff
@@ -405,6 +409,12 @@ reader_permit::resource_units reader_permit::consume_resources(reader_resources
     return resource_units(*this, res);
 }
 
+future<reader_permit::resource_units> reader_permit::request_memory(size_t memory) {
+    return _impl->request_memory(memory).then([this, memory] {
+        return resource_units(*this, {0, ssize_t(memory)});
```

Comment:
Minor: should pass *this to _impl->request_memory to avoid the continuation.

Not too important since in the regular case all those futures are ready. Can do it as a follow up.

Reply:
Actually, we can't avoid the continuation, the call goes into the semaphore.